### PR TITLE
fix issue for self cycles

### LINKED ISSUE #418: Auto-tick PR stopped working for tbb-feedstock
PR is not created automatically for https://github.com/01org/tbb/tree/2019_U3
Re-opened https://github.com/conda-forge/tbb-feedstock/issues/29

Thanks!

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes #418 
closes https://github.com/conda-forge/tbb-feedstock/issues/29

### CODE REVIEW

**Comment by scopatz** [comment]
LGTM feel free to merge after CI

**Comment by CJ-Wright** [comment]
This is what I get for doing this in a naked text editor.

File: conda_forge_tick/migrators.xsh [hunk 1]
```diff
@@ -165,7 +165,8 @@ class Migrator:
 
         """
         top_level = set(
-            node for node in graph if not list(graph.predecessors(node)))
+            node for node in graph if not list(graph.predecessors(node)) 
+                                      or list(graph.predecessors(node) == [node])
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]:
```suggestion
                                      or list(graph.predecessors(node)) == [node]
```

File: conda_forge_tick/migrators.xsh [hunk 2]
```diff
@@ -165,7 +165,7 @@ class Migrator:
 
         """
         top_level = set(
-            node for node in graph if not list(graph.predecessors(node)))
+            node for node in graph if not list(graph.predecessors(node)) or list(graph.predecessors(node)) == [node]
```
Comment (scopatz) [inline conda_forge_tick/migrators.xsh]: missing closing paren

**Comment by CJ-Wright** [comment]
Thank you!

**Comment by anton-malakhov** [comment]
Thank you!

**Comment by jakirkham** [comment]
Thanks all 😀

**Pull request merged**